consensus/state.go

```diff
@@ -1306,6 +1306,22 @@ func (cs *ConsensusState) addProposalBlockPart(height int64, part *types.Part, v
 		}
 		// NOTE: it's possible to receive complete proposal blocks for future rounds without having the proposal
 		cs.Logger.Info("Received complete proposal block", "height", cs.ProposalBlock.Height, "hash", cs.ProposalBlock.Hash())
+
+		// Update ValidBlock
+		prevotes := cs.Votes.Prevotes(cs.Round)
+		blockID, ok := prevotes.TwoThirdsMajority()
+		if ok && !blockID.IsZero() && (cs.ValidRound < cs.Round) {
```

Comment:
why `ValidRound < cs.Round` and not `ValidRound <= cs.Round` ?

Reply:
oh nvm - thats just saying if we have a ValidBlock from a previous round, we can update to the one for the current round